clang/lib/Driver/ToolChains/HIPUtility.cpp

```diff
@@ -463,10 +463,11 @@ void HIP::constructGenerateObjFileFromHIPFatBinary(
 
   Objf << ObjBuffer;
 
-  ArgStringList McArgs{"-triple", Args.MakeArgString(HostTriple.normalize()),
+  ArgStringList McArgs{"-target", Args.MakeArgString(HostTriple.normalize()),
                        "-o",      Output.getFilename(),
-                       McinFile,  "--filetype=obj"};
-  const char *Mc = Args.MakeArgString(TC.GetProgramPath("llvm-mc"));
+                       "-x",      "assembler",
+                       ObjinFile, "-c"};
+  const char *Mc = Args.MakeArgString(TC.GetProgramPath("clang"));
```

Comment:
Shouldn't assume the binary name is clang, other places seem to be doing something like this:

`TCArgs.MakeArgString(getToolChain().GetProgramPath(getShortName()))`

Reply:
Why would it be called anything but `clang`? `getShortName` just returns the name of the Tool, in this case it would return stuff like `hipspv-link` or `amdgcn-link`.

Reply:
Because sometimes there's a version suffix (e.g. clang-19), and some distributions add on random prefixes or suffixes (e.g. amdclang)

Reply:
I don't think there's an easy way to do this, since it would require knowing `argv[0]` basically. There's some GNU specific hacks to get that, but I don't think it's critical that the clang we invoke here is *the* amdclang, since it's just doing a simple assembler to object file job.

Reply:
But the toolchain tracked the name of the current clang? Really you want to find the current binary

`I don't think it's critical that the clang we invoke here is the amdclang`

It's critical to find the exact clang that you are running

Reply:
Turns out that we have `ClangExecutable` in the Driver.